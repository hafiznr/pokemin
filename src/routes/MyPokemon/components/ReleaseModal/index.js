/** @jsxRuntime classic */
/** @jsx jsx */
import { 
  Button, 
  Modal, 
  ModalContent, 
  ModalHeader, 
  ModalBody, 
  ModalOverlay,
  ButtonGroup
} from '@chakra-ui/react';
import { jsx, css } from '@emotion/react';

const ReleaseModal = ({
  isOpen,
  pokemon = '',
  image,
  onClose,
  onSubmit
}) => {
  const center = css({
    textAlign: 'center'
  });

  const imgStyle = css({
    display: 'inline'
  })

  return (
    <Modal 
      isOpen={isOpen} 
      onClose={onClose} 
      closeOnOverlayClick={false} 
      size="xs"
      isCentered
    >
      <ModalOverlay />
      
      <ModalContent>
        <ModalHeader>
          <div css={center}>Are you sure you want to release {pokemon}?</div>
        </ModalHeader>
        <ModalBody>
          <div css={center}>
            <img css={imgStyle} src={image} alt={pokemon} loading="lazy"/>
            
            <ButtonGroup spacing="4" width="100%">
              <Button onClick={onClose} width="100%">Cancel</Button>
              <Button onClick={onSubmit} colorScheme="red" width="100%">Release</Button>
            </ButtonGroup>
          </div>
        </ModalBody>
      </ModalContent>
    </Modal>
  )
}

export default ReleaseModal;