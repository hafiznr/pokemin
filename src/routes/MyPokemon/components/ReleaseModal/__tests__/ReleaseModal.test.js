import { render } from '@testing-library/react';
import ReleaseModal from '../';

test('renders ReleaseModal', () => {
  const { container } = render(<ReleaseModal/>);

  expect(container).toMatchSnapshot();
});