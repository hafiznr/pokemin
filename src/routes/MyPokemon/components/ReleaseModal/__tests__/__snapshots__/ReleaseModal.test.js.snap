// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ReleaseModal 1`] = `<div />`;
